### ISSUE TITLE
Invalid completion items when using dots in completion item label

### ISSUE THREAD

**Issue by lmvco** [microsoft/monaco-editor#2136]
**monaco-editor version:** 0.20.0
**Browser:** Chrome
**OS:** Windows
**Playground code that reproduces the issue:**

```
function getSuggestions(range) {
    return [
        {
            label: 'parameters.In1',
            kind: monaco.languages.CompletionItemKind.Field,
            insertText: 'parameters.In1',
            insertTextRules: null,
            range: range
        }
    ];
}


monaco.languages.registerCompletionItemProvider('javascript', {
    provideCompletionItems: function(model, position) {
        return {
            suggestions: getSuggestions(null)
        };
    }
});

monaco.editor.create(document.getElementById("container"), {
    language: "javascript"
});
```

I was only able to reproduce this when registering a completion item provider which return completion items with dots in the label/insertText.

Repro steps:
1. Execute the code above
2. Type `param` and choose the option `parameters.In1`
3. Type again `type`
Result: The completion menu shows an option "parameters" which was not returned by my provider neither exists in the Javascript language.

See the gif below:
![custom-js-autocomplete3](https://user-images.githubusercontent.com/1884391/93500095-c13a4780-f90b-11ea-9d43-8d8b01eb7b62.gif)

**Comment by spahnke** [comment]
It's even easier to reproduce. Just type `foo.` into the playground editor directly after the sample code and trigger suggestions https://microsoft.github.io/monaco-editor/playground.html. Since you can't infer anything there (`foo` is of type `any` as far as the language server is concerned) the editor shows you words used in the document to be at least a little helpful.
![image](https://user-images.githubusercontent.com/25029871/93503824-e8e3dc80-f918-11ea-87aa-84c447046bc2.png)
I thought these were word-based/quick suggestion but neither `wordBasedSuggestions: false` nor `quickSuggestions: false` make a difference so I'm actually not sure anymore.

**Comment by spahnke** [comment]
>I thought these were word-based/quick suggestion but neither wordBasedSuggestions: false nor quickSuggestions: false make a difference so I'm actually not sure anymore.

Ah, that's because of #1980

**Comment by orta** [comment]
Yep, just confirming this is a response from the typescript side:

```json
    {"seq":0,"type":"response","command":"completionInfo","request_seq":24,"success":true,"performanceData":{"updateGraphDurationMs":5.120795011520386},"body":{"isGlobalCompletion":false,"isMemberCompletion":true,"isNewIdentifierLocation":false,"entries":[{"name":"alert","kind":"warning","kindModifiers":"","sortText":"7"},{"name":"foo","kind":"warning","kindModifiers":"","sortText":"7"},{"name":"hello","kind":"warning","kindModifiers":"","sortText":"7"}]}}
```

If someone wants to solve this they'll need to add a flag to the `UserPreferences` type in TypeScript which surpasses passing file identifiers, get that shipped in a TS release and then have monaco-typescript support setting it if `wordBasedSuggestions` is false
